scripts/linking.py

```diff
@@ -534,8 +542,23 @@ def main(medmentions_path: str,
     else:
         thresholds = [float(x) for x in thresholds.split(",")]
 
-    # only evaluate on the dev examples for now because we don't have a trained model
-    eval_candidate_generation_and_linking(dev_examples, umls_concept_dict_by_id, candidate_generator, linking_classifier, k_list, thresholds, use_gold_mentions, spacy_model)
+    if generate_linker_data:
+        examples_list = [train_examples, dev_examples, test_examples]
+        filenames = [f'{model_path}/train.jsonl', f'{model_path}/dev.jsonl', f'{model_path}/test.jsonl']
+        for examples, filename in zip(examples_list, filenames):
+            supervised_data = eval_candidate_generation_and_linking(examples, umls_concept_dict_by_id, candidate_generator, linking_classifier, k_list, thresholds,
+                                                                    use_gold_mentions, spacy_model, generate_linker_data)
+            with jsonlines.open(filename, 'w') as f:
+                for d in supervised_data:
+                    f.write(d)
+    else:
+        print('Results on the DEV set')
+        eval_candidate_generation_and_linking(dev_examples, umls_concept_dict_by_id, candidate_generator, linking_classifier, k_list, thresholds,
+                                              use_gold_mentions, spacy_model, generate_linker_data)
+
+        print('Results on the TEST set')
```

Comment:
Let's remove this so we don't peek at the test set by default when running the script.

Reply:
ok